ema_workbench/examples/example_mesa.py

```diff
@@ -0,0 +1,250 @@
+        susceptible_neighbors = [
+            agent
+            for agent in self.model.grid.get_cell_list_contents(neighbors_nodes)
+            if agent.state is State.SUSCEPTIBLE
+        ]
+        for a in susceptible_neighbors:
+            if self.random.random() < self.virus_spread_chance:
+                a.state = State.INFECTED
+
+    def try_gain_resistance(self):
+        if self.random.random() < self.gain_resistance_chance:
+            self.state = State.RESISTANT
+
+    def try_remove_infection(self):
+        # Try to remove
+        if self.random.random() < self.recovery_chance:
+            # Success
+            self.state = State.SUSCEPTIBLE
+            self.try_gain_resistance()
+        else:
+            # Failed
+            self.state = State.INFECTED
+
+    def try_check_situation(self):
+        if (self.random.random() < self.virus_check_frequency) and (self.state is State.INFECTED):
+            self.try_remove_infection()
+
+    def step(self):
+        if self.state is State.INFECTED:
+            self.try_to_infect_neighbors()
+        self.try_check_situation()
+
+
+# Setting up the model as a function
+def model_virus_on_network(
+    num_nodes=1,
+    avg_node_degree=1,
+    initial_outbreak_size=1,
+    virus_spread_chance=1,
+    virus_check_frequency=1,
+    recovery_chance=1,
+    gain_resistance_chance=1,
+    steps=10,
+):
+    # Initialising the model
+    virus_on_network = VirusOnNetwork(
+        num_nodes=num_nodes,
+        avg_node_degree=avg_node_degree,
+        initial_outbreak_size=initial_outbreak_size,
+        virus_spread_chance=virus_spread_chance,
+        virus_check_frequency=virus_check_frequency,
+        recovery_chance=recovery_chance,
+        gain_resistance_chance=gain_resistance_chance,
+    )
+
+    # Run the model steps times
+    virus_on_network.run_model(steps)
+
+    # Get model outcomes
+    outcomes = virus_on_network.datacollector.get_model_vars_dataframe()
+
+    # Return model outcomes
+    return {
+        "TIME": list(range(steps + 1)),
+        "Infected": outcomes["Infected"].tolist(),
+        "Susceptible": outcomes["Susceptible"].tolist(),
+        "Resistant": outcomes["Resistant"].tolist(),
+    }
+
+
+# Instantiate and pass the model
+model = ReplicatorModel("VirusOnNetwork", function=model_virus_on_network)
+
+# Define model parameters and their ranges to be sampled
+model.uncertainties = [
+    IntegerParameter("num_nodes", 10, 100),
+    IntegerParameter("avg_node_degree", 2, 8),
+    RealParameter("virus_spread_chance", 0.1, 1),
+    RealParameter("virus_check_frequency", 0.1, 1),
+    RealParameter("recovery_chance", 0.1, 1),
+    RealParameter("gain_resistance_chance", 0.1, 1),
+]
+
+# Define model parameters that will remain constant
+model.constants = [Constant("initial_outbreak_size", 1), Constant("steps", 30)]
+
+# Define model outcomes
+model.outcomes = [
+    TimeSeriesOutcome("TIME"),
```

Comment:
why do you include time here?

and since you use a replicator model, these should all be replaced with ArrayOutcomes

Reply:
I have a dim recollection of some EMA plotting functions breaking (or giving a warning?) if there is no TIME dimension, so this was probably included to preclude that.

Reply:
With ArrayOutcomes these will break anyway. Also, the time problem is now fixed and only will give a warning.